tests/phpunit/elementor/core/page-assets/test-loader.php

```diff
@@ -0,0 +1,202 @@
+		$this->assertContains( 'e-script-default-enqueue', $wp_scripts->queue, 'e-script-default-enqueue was not properly enqueued.' );
+
+		// When "optimized assets loading" mode is active, assets should be loaded only when enabled.
+		$this->elementor()->experiments->set_feature_default_state( 'e_optimized_assets_loading', Experiments_Manager::STATE_ACTIVE );
+
+		$this->add_testing_assets( 'e-style-dynamic-enqueue', 'e-script-dynamic-enqueue' );
+
+		$assets_loader->enqueue_assets();
+
+		// Making sure that the assets are not enqueued before they are enabled.
+		$this->assertNotContains( 'e-style-dynamic-enqueue', $wp_styles->queue, 'e-style-dynamic-enqueue should not be enqueued before it\'s enabled.' );
+		$this->assertNotContains( 'e-script-dynamic-enqueue', $wp_scripts->queue, 'e-script-dynamic-enqueue should not be enqueued before it\'s enabled.' );
+
+		$assets_loader->enable_assets( [
+			'styles' => ['e-style-dynamic-enqueue'],
+			'scripts' => ['e-script-dynamic-enqueue'],
+		] );
+
+		$assets_loader->enqueue_assets();
+
+		$this->assertContains( 'e-style-dynamic-enqueue', $wp_styles->queue, 'e-style-dynamic-enqueue was not properly enqueued.' );
+		$this->assertContains( 'e-script-dynamic-enqueue', $wp_scripts->queue, 'e-script-dynamic-enqueue was not properly enqueued.' );
+	}
+
+	public function test_register_assets() {
+		global $wp_styles;
+		global $wp_scripts;
+
+		// Creating a new instance to get only the initial assets that exist in the assets loader (without the testing assets that were added by the other testing methods).
+		$new_assets_loader_instance = new Assets_Loader();
+
+		$current_assets = $new_assets_loader_instance->get_assets();
+
+		// Deregister the current assets.
+		foreach ( $current_assets as $assets_type => $assets_type_data ) {
+			foreach ( $assets_type_data as $asset_name => $asset_data ) {
+				if ( 'scripts' === $assets_type ) {
+					wp_deregister_script( $asset_name );
+					$this->assertArrayNotHasKey( $asset_name, $wp_scripts->registered, 'script asset should not be registered.' );
+				} else {
+					wp_deregister_style( $asset_name );
+					$this->assertArrayNotHasKey( $asset_name, $wp_styles->registered, 'style asset should not be registered.' );
+				}
+			}
+		}
+
+		// Creating additional instance in order to trigger the private 'register_assets' method that initiated by the Assets Loader constructor.
+		new Assets_Loader();
+
+		// Making sure that the additional assets loader instance has registered all the assets that were de-registered above.
+		foreach ( $current_assets as $assets_type => $assets_type_data ) {
+			foreach ( $assets_type_data as $asset_name => $asset_data ) {
+				if ( 'scripts' === $assets_type ) {
+					$this->assertArrayHasKey( $asset_name, $wp_scripts->registered, 'script asset was not properly registered.' );
+				} else {
+					$this->assertArrayHasKey( $asset_name, $wp_styles->registered, 'style asset was not properly registered.' );
+				}
+			}
+		}
```

Comment:
Split to a different case

Reply:
As agreed, keeping it as it is, because everything is part of the same single test.